pod/perlguts.pod

```diff
@@ -1023,6 +1023,62 @@ as any other SV.
 
 For more information on references and blessings, consult L<perlref>.
 
+=head2 I/O Handles
+
+Like AVs and HVs, IO objects are another type of non-scalar SV which
+may contain input and output L<PerlIO|perlapio> objects or a C<DIR *>
+from opendir().
+
+You can create a new IO object:
+
+    IO*  newIO();
+
+The IO itself has little distinct function from other SVs, it's just a
+store for handles and information about those handles.
+
+Unlike other SVs, a new IO object is automatically blessed into the
+L<IO::File> class.
```

Comment:
Way out of my depth here but isn't this only true for file objects? (as opposed to dirhandles, sockets, etc)

Reply:
https://github.com/Perl/perl5/blob/a473ef8bc0d67b85bc4982944d05f235fc906aae/sv.c#L1459-L1471

It happens when the SV is upgraded to an IO object, before any of the IO slots are populated

```
$ perl -MDevel::Peek -e 'opendir my $f, "." or die; Dump(*$f{IO})'
SV = IV(0x55d2aaedfb38) at 0x55d2aaedfb48
  REFCNT = 1
  FLAGS = (TEMP,ROK)
  RV = 0x55d2aae9a4a0
  SV = PVIO(0x55d2aaec3630) at 0x55d2aae9a4a0
    REFCNT = 2
    FLAGS = (OBJECT)
    STASH = 0x55d2aaebd100      "IO::File"
    IFP = 0x0
    OFP = 0x0
    DIRP = 0x55d2aaf0e950
    LINES = 0
    PAGE = 0
    PAGE_LEN = 60
    LINES_LEFT = 0
    TOP_GV = 0x0
    FMT_GV = 0x0
    BOTTOM_GV = 0x0
    TYPE = '\0'
    FLAGS = 0x0
```